### ISSUE TITLE
CircleCI legacy 32-bit build fails CPU detection

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1888]
```
Makefile.legacy:686: recipe for target 'linux-x86-sse2' failed

make: *** [linux-x86-sse2] Error 2

Sorry, SSE2 is required for this build
```
I have verified it works fine with autoconf.

**Comment by magnumripper** [comment]
Apparently I tested all variants except exactly SSE2. I can reproduce on super:
```
$ ./configure CC="gcc -m32 -msse2" --disable-native-tests --host=i686-linux-gnu && make -s clean && make -sj32
(...)
$ ../run/john -test -form:nt
Sorry, SSE2 is required for this build
```

**Comment by magnumripper** [comment]
Fixed in 258b18a

**Comment by magnumripper** [comment]
CircleCI confirmed its fine now!
